src/aliceVision/sfm/BundleAdjustmentPanoramaCeres.cpp

```diff
@@ -0,0 +1,882 @@
+  }
+  else
+  {
+    linearSolverType = ceres::DENSE_SCHUR;
+    ALICEVISION_LOG_WARNING("BundleAdjustment[Ceres]: no sparse BA available, fallback to dense BA.");
+  }
+}
+
+bool BundleAdjustmentPanoramaCeres::Statistics::exportToFile(const std::string& folder, const std::string& filename) const
+{
+  std::ofstream os;
+  os.open((fs::path(folder) / filename).string(), std::ios::app);
+
+  if(!os.is_open())
+  {
+    ALICEVISION_LOG_DEBUG("Unable to open the Bundle adjustment statistics file: '" << filename << "'.");
+    return false;
+  }
+
+  os.seekp(0, std::ios::end); // put the cursor at the end
+
+  if(os.tellp() == std::streampos(0)) // 'tellp' return the cursor's position
+  {
+    // if the file does't exist: add a header.
+    os << "Time/BA(s);RefinedPose;ConstPose;IgnoredPose;"
+          "RefinedPts;ConstPts;IgnoredPts;"
+          "RefinedK;ConstK;IgnoredK;"
+          "ResidualBlocks;SuccessIteration;BadIteration;"
+          "InitRMSE;FinalRMSE;"
+          "d=-1;d=0;d=1;d=2;d=3;d=4;"
+          "d=5;d=6;d=7;d=8;d=9;d=10+;\n";
+  }
+
+  std::map<EParameter, std::map<EParameterState, std::size_t>> states = parametersStates;
+  std::size_t posesWithDistUpperThanTen = 0;
+
+  for(const auto& it : nbCamerasPerDistance)
+    if (it.first >= 10)
+      posesWithDistUpperThanTen += it.second;
+
+  os << time << ";"
+     << states[EParameter::POSE][EParameterState::REFINED]  << ";"
+     << states[EParameter::POSE][EParameterState::CONSTANT] << ";"
+     << states[EParameter::POSE][EParameterState::IGNORED]  << ";"
+     << states[EParameter::INTRINSIC][EParameterState::REFINED]  << ";"
+     << states[EParameter::INTRINSIC][EParameterState::CONSTANT] << ";"
+     << states[EParameter::INTRINSIC][EParameterState::IGNORED]  << ";"
+     << nbResidualBlocks << ";"
+     << nbSuccessfullIterations << ";"
+     << nbUnsuccessfullIterations << ";"
+     << RMSEinitial << ";"
+     << RMSEfinal << ";";
+
+     for(int i = -1; i < 10; ++i)
+     {
+       auto cdIt = nbCamerasPerDistance.find(i);
+       if(cdIt != nbCamerasPerDistance.end())
+        os << cdIt->second << ";";
+       else
+         os << "0;";
+     }
+
+     os << posesWithDistUpperThanTen << ";\n";
+
+  os.close();
+  return true;
+}
+
+void BundleAdjustmentPanoramaCeres::Statistics::show() const
+{
+  std::map<EParameter, std::map<EParameterState, std::size_t>> states = parametersStates;
+  std::stringstream ss;
+
+  if(!nbCamerasPerDistance.empty())
+  {
+    std::size_t nbCamNotConnected = 0;
+    std::size_t nbCamDistEqZero = 0;
+    std::size_t nbCamDistEqOne = 0;
+    std::size_t nbCamDistUpperOne = 0;
+
+    for(const auto & camdistIt : nbCamerasPerDistance)
+    {
+      if(camdistIt.first < 0)
+        nbCamNotConnected += camdistIt.second;
+      else if(camdistIt.first == 1)
+        nbCamDistEqZero += camdistIt.second;
+      else if(camdistIt.first == 1)
+        nbCamDistEqOne += camdistIt.second;
```

Comment:
As pointed out by codacy, these two conditions are the same. Which one should be removed or what condition should be changed?

Reply:
Yes, obviously it should be:
```
      if(camdistIt.first < 0)
        nbCamNotConnected += camdistIt.second;
      else if(camdistIt.first == 0)  //<<< Should be zero here!
        nbCamDistEqZero += camdistIt.second;
      else if(camdistIt.first == 1)
        nbCamDistEqOne += camdistIt.second;
```